markdown/htmlparser.py

```diff
@@ -26,6 +26,10 @@
 parser.piclose = re.compile(r'\?>')
 # Monkeypatch HTMLParser to only recognize entity references with a closing semicolon.
 parser.entityref = re.compile(r'&([a-zA-Z][-.a-zA-Z0-9]*);')
+# Monkeypatch HTMLParser to no longer support partial entities. We are always feeding a complete block,
```

Comment:
Is monkey patching the only way to address this? I'm worried that there may be side effects if using markdown with other libraries that rely on `html.parser`. I'm curious if there is some cleaner way to do this without altering the builtin library..

Reply:
So, I was looking into this. What if we modify a copy of the module instead of the original?

```py
>>> import importlib
>>> import sys
>>> spec = importlib.util.find_spec('html.parser')
>>> monkey = importlib.util.module_from_spec(spec)
>>> spec.loader.exec_module(monkey)
>>> sys.modules['monkey'] = monkey
>>> monkey.entityref = 'whatever'
>>> from html import parser
>>> parser.entityref
re.compile('&([a-zA-Z][-.a-zA-Z0-9]*)[^a-zA-Z0-9]')
>>> monkey.entityref
'whatever'
```

This way we could use our monkey patched version, but not modify the original. What do you think?

Reply:
> I'm worried that there may be side effects if using markdown with other libraries that rely on `html.parser`.

That's a valid concern. I had been avoiding the issue because I didn't want to maintain a copy of the lib. However...

> we could use our monkey patched version, but not modify the original.

That works around my concern quite well. I like it.

Reply:
Change has been made in efa36c8.